updates_and_signals/safe_message_handlers/src/test/workflows.test.ts

```diff
@@ -0,0 +1,90 @@
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { before, describe, it } from 'mocha';
+import { bundleWorkflowCode, WorkflowBundleWithSourceMap, DefaultLogger, Runtime, Worker } from '@temporalio/worker';
+import * as activities from '../activities';
+import {
+  clusterManagerWorkflow,
+  assignNodesToJobUpdate,
+  startClusterSignal,
+  shutdownClusterSignal,
+  deleteJobUpdate,
+  getClusterStatusQuery,
+} from '../workflows';
+import { nanoid } from 'nanoid';
+import assert from 'assert';
+
+const taskQueue = 'test' + new Date().toLocaleDateString('en-US');
+
+describe('cluster manager', function () {
+  this.timeout(10000);
+  let worker: Worker;
+  let env: TestWorkflowEnvironment;
+  let workflowBundle: WorkflowBundleWithSourceMap;
+
+  before(async function () {
+    Runtime.install({ logger: new DefaultLogger('WARN') });
+    env = await TestWorkflowEnvironment.createLocal();
+
+    workflowBundle = await bundleWorkflowCode({
+      workflowsPath: require.resolve('../workflows'),
+      logger: new DefaultLogger('WARN'),
+    });
+  });
+
+  beforeEach(async function () {
+    worker = await Worker.create({
+      connection: env.nativeConnection,
+      workflowBundle,
+      activities,
+      taskQueue,
+    });
+  });
+
+  after(async function () {
+    await env.teardown();
+  });
+
+  it('successfully completes a session', async function () {
+    await worker.runUntil(async function () {
+      const workflow = await env.client.workflow.start(clusterManagerWorkflow, {
+        args: [{}],
```

Comment:
As said before, set default arg value at the function level instead of explicitly passing this empty object.